docs/Roadmap.md

```diff
@@ -0,0 +1,112 @@
+    - [General](#general)
+    - [Probes](#probes)
+    - [Notify](#notify)
+
+
+## Product Principles
+EaseProbe tries to do two things but do them extremely well - **Probe** and **Notify**.
+
+1. **Probe and Notify**. Aims to provide the ability to easily **probe** servers, services and API endpoints reliably and **notify** on status updates.
+
+2. **Open & Extensible**. It aims to be **Open** and follow Open standards that allow it to be integrated into **extensible-development** platforms (such as EaseMesh).
+
+3. **Cloud Native**. It is designed to be **cloud-native** compliant. It is scalable, resilient, manageable, and observable and it's easy to integrate with cloud-native architectures of any type.
+
+4. **Predictability & Reliability**. EaseProbe is designed to follow best practices allowing predictable and reliable operations. Even when EaseProbe fails, it aims to fail in a predictable way which allows for easy and speedy troubleshooting of problems.
+
+
+## Features
+The project principles, of EaseProbe' features are separated into two main categories: probe-specific & notify-specific.
+
+### Probe specific
+* HTTP for testing connectivity and validity of responses
+* TCP for testing connectivity
+* Shell for executing custom probe scripts
+* SSH for remote ssh commands
+* Host for CPU, Memory, Disk usage metrics
+* Clients for Redis, MySQL, MongoDB, PostgreSQL, Kafka, Zookeeper
+
+### Notify specific
+* Mail notification
+* AWS SNS notification
+* Log files 
+* Slack
+* Discord
+* Telegram
+* WeChat Work
+* Lark
+* DingTalk
+* Twilio
+* Nexmo
+* YunPian
+
+... with new notification backends been added constantly.
+
+## Roadmap 2022
+Some of the features that are planned/considered for 2022 can be broken down into three categories: *General*, *Probe*, *Notify*
+
+
+### General
+* Work on detailed documentation
+* support for common daemon features
+  * [x] megaease/easeprobe#75 add `daemon()` & `/var/run/easeprobe.pid` support
+  * [x] megaease/easeprobe#75 add SIGHUP, and ensure it closes and re-opens of logfile to allow for `easeprobe.log` rotation
+  * add syslog support eg `logfile: syslog`
+* Support for common `timeformat`, use standard timezone and `strftime` conversions, eg `timezone: [UTC|local|Europe/Athens]`, `timeformat: %F %R:%S UTC`
+* Add opt-out options where appropriate
+  * [x] megaease/easeprobe#75 add opt-out option for `logfile` option 
+  * Add opt-out option for SLA data persistence `data: false`
+  * Make historical data configurable `history: false` and avoid creating backups of statistics
+
+### Probes
+* integrate automatic service discovery which includes probing details
+* add plain old icmp ping probe (or read next)
+* shell probe command must allow allocation or not of `stdout`/`stderr` or redirection of them (eg right now we are forced to create wrapper scripts to achieve this) This will also make ping by shell usable
+* Add support to define desired notification on probes???
```

Comment:
how about introducing a group concept? something like:

```yaml
GroupA:  
   http:
   server:
   client:
   notify:
      discord:
      slack:
GroupB:
   ....
```

Reply:
This could work i think, the functionality i was aiming was more on the lines of:
i have two probes, but one of them needs priority in resolving any issues. For that reason i configure say SMS notify, but for the other i dont care that much so i want it to go to the general chat of a discord server.

With the grouping you describe we will end up with two groups with 1 notification each (?).

Reply:
Thanks for the clarification, it makes sense to me. 

And yes, your understanding is right - two groups with 1 notification for each. **Basically, the entity-relationship is** -  **the probes are grouped by notfication.** 

And,  to cover all of the scenarios, the best model is the `n:m` relationship between probe and notification,  but this would introduce the complexities of configuration - we need an extra configuration to describe the relationship.

something like.

```yaml
relationsip:
- probe_id_1 : notification_id_1
- probe_id_2 : notification_id_1
- probe_id_1 : notification_id_2
....
```

Reply:
Yes i was facing a similar problem when i tried to _describe_ it with some fake configuration examples just to see what made more sense but it gets confusing very easily. 

The current configuration layout for easeprobe is a beauty i would hate to make it confusing just to support these features, but maybe we can start with a simple support for it and expand it later as we get more data and use-cases? 

stage1: ability to point to a single notify end point based on the current name
stage2: ability to support more elaborate groupings
stage3: full relationships support between probes and notify endpoints

Just an idea